core/src/test/scala/akka/persistence/r2dbc/query/EventsBySliceSpec.scala

```diff
@@ -168,6 +214,64 @@ class EventsBySliceSpec
         assertFinished(result)
       }
 
+      "handle more events with same timestamp than buffer size" in new Setup {
+        val queryWithSmallBuffer = PersistenceQuery(testKit.system) // buffer size = 4
+          .readJournalFor[R2dbcReadJournal]("akka.persistence.r2dbc-small-buffer.query")
+        persister ! PersistAll((1 to 10).map(i => s"e-$i").toList)
+        persister ! Ping(probe.ref)
+        probe.expectMessage(Done)
+        val result: TestSubscriber.Probe[EventEnvelope[String]] =
+          doQuery(entityType, slice, slice, NoOffset, queryWithSmallBuffer)
+            .runWith(sinkProbe)
+            .request(11)
+        for (i <- 1 to 10) {
+          result.expectNext().event shouldBe s"e-$i"
+        }
+        assertFinished(result)
+      }
+
+      "handle more events with same timestamp than buffer size, with overlapping seq numbers" in {
+        val entityType = nextEntityType()
+        val pid1 = nextPid(entityType)
+        val pid2 = nextPid(entityType)
+        val pid3 = nextPid(entityType)
+        val slice1 = query.sliceForPersistenceId(pid1)
+        val slice2 = query.sliceForPersistenceId(pid2)
+        val slice3 = query.sliceForPersistenceId(pid3)
+        val slices = Seq(slice1, slice2, slice3)
+
+        val timestamp = InstantFactory.now()
+        writeEvent(slice1, pid1, 1L, timestamp, "A1")
+        writeEvent(slice1, pid1, 2L, timestamp, "A2")
+        writeEvent(slice1, pid1, 3L, timestamp, "A3")
+        writeEvent(slice1, pid1, 4L, timestamp, "A4")
+        writeEvent(slice1, pid1, 5L, timestamp, "A5")
+        writeEvent(slice1, pid1, 6L, timestamp, "A6")
+        writeEvent(slice2, pid2, 3L, timestamp, "B3")
+        writeEvent(slice2, pid2, 4L, timestamp, "B4")
+        writeEvent(slice3, pid3, 3L, timestamp, "C3")
+
+        val queryWithSmallBuffer = PersistenceQuery(testKit.system) // buffer size = 4
+          .readJournalFor[R2dbcReadJournal]("akka.persistence.r2dbc-small-buffer.query")
+
+        val sinkProbe = TestSink[EventEnvelope[String]]()
+
+        val result: TestSubscriber.Probe[EventEnvelope[String]] =
+          doQuery(entityType, slices.min, slices.max, NoOffset, queryWithSmallBuffer)
+            .runWith(sinkProbe)
+            .request(10)
+
+        def take(n: Int): Set[String] =
+          (1 to n).map(_ => result.expectNext().event).toSet
+
+        take(2) shouldBe Set("A1", "A2")
+        take(3) shouldBe Set("A3", "B3", "C3")
+        take(2) shouldBe Set("A4", "B4")
+        take(2) shouldBe Set("A5", "A6")
```

Comment:
Postgres does return the events with the same seq number in the inserted order, but probably shouldn't rely on that for tests.